### PR TITLE
Fixed errors in cleanupTarchives.pl tool

### PR DESCRIPTION
**Pull request by cmadjar**
### Description

This PR fixes and cleans up the tool `cleanupTarchives.pl` and document it with Perldoc at the same time.

Before the fixes, got the following errors:
```
defined(@array) is deprecated at cleanupTarchives.pl line 54.
    (Maybe you should just omit the defined()?)
Global symbol "$this" requires explicit package name at cleanupTarchives.pl line 71.
Global symbol "$this" requires explicit package name at cleanupTarchives.pl line 74.
Execution of cleanupTarchives.pl aborted due to compilation errors.
```

### This fixes

- [x] Redmine https://redmine.cbrain.mcgill.ca/issues/15041

### How to test this

- [ ] Read the documentation to make sure it all makes sense
- [ ] Create with the `touch` command a few dummy files in `/data/project/data/tarchive` that matches the filename of one or several of the DICOM archives that you have in the year subfolders (to create false duplicate)
- [ ] Run `cleanupTarchives.pl` and make sure the files you created with the `touch` command are no longer in `/data/project/data/tarchive`

### CODE REVIEW

File: tools/cleanupTarchives.pl [hunk 1]
```diff
@@ -91,34 +129,38 @@
 ####     Main program     ####
 ##############################
 
-# Get tarchives list from the database and stores ArchiveLocation and md5sumArchive informations in a hash.
-## ArchiveLocation will be the key of the hash
-## md5sumArchive will be the value of the hash
-my ($tarchivesList_db)  = &selectTarchives($dbh, $tarchiveLibraryDir);
+# Get the list of DICOM archives the tarchive table of the database and stores
```
Comment (nicolasbrossard) [inline tools/cleanupTarchives.pl]: __Gets__ the list of DICOM archives __from__ the tarchive table of the database and...

File: tools/cleanupTarchives.pl [hunk 2]
```diff
@@ -221,11 +291,20 @@ sub selectTarchives {
...
+
+=head3 getTarchiveBasename($tarchive)
+
+Function that will determine the DICOM archive basename from the C<ArchiveLocation>
+stored in the database. It will, among other things, get rid of the C<_digit part>
+that was inserted in the past by the C<tarchiveLoader>.
+
+INPUT: C<ArchiveLocation> that was stored in the C<tarchive> table of the database.
+
+RETURNS: the DICOM archive basename to use to look for duplicate DICOM archives in
```
Comment (nicolasbrossard) [inline tools/cleanupTarchives.pl]: ...basename to use __when looking__ for duplicate DICOM...

File: tools/cleanupTarchives.pl [hunk 3]
```diff
@@ -243,21 +322,36 @@ sub getTarchiveBasename {
...
-- Outputs: - Undef => if no tarchive matching the file stored in the database could be found in the filesystem.
-           - @duplicateTarFiles and $realTarFileFound => if the file stored in the database matches one file of the tarFileList present in the filesystem. 
+
+=head3 identifyDuplicates($tarchive_db, $tarchivesList_db, $tarFileList)
+
+Function that will identify the duplicate DICOM archives present in the filesystem.
+
+INPUTS:
+  - $tarchive_db     : DICOM archive file stored in the database's C<tarchive> table
+  - $tarchivesList_db: hash with the list of DICOM archives location stored in the
```
Comment (nicolasbrossard) [inline tools/cleanupTarchives.pl]: ..DICOM archives __locations__...

File: tools/cleanupTarchives.pl [hunk 4]
```diff
@@ -243,21 +322,36 @@ sub getTarchiveBasename {
...
+                       database (keys of the hash) and their corresponding md5sum
+                       (values of the hash)
+  - tarFileList      : list of DICOM archives found in the filesystem that matches
```
Comment (nicolasbrossard) [inline tools/cleanupTarchives.pl]: ...in the filesystem that __match__ the basename...

File: tools/cleanupTarchives.pl (hunk 1 above)
Comment (cmadjar) [reply tools/cleanupTarchives.pl]: done

File: tools/cleanupTarchives.pl (hunk 2 above)
Comment (cmadjar) [reply tools/cleanupTarchives.pl]: done

File: tools/cleanupTarchives.pl (hunk 3 above)
Comment (cmadjar) [reply tools/cleanupTarchives.pl]: done

File: tools/cleanupTarchives.pl (hunk 4 above)
Comment (cmadjar) [reply tools/cleanupTarchives.pl]: done

**Pull request merged**